packages/mobile/src/send/ValidateRecipientAccount.tsx

```diff
@@ -0,0 +1,301 @@
+}
+
+const mapDispatchToProps = {
+  validateRecipientAddress,
+}
+
+const mapStateToProps = (state: RootState, ownProps: NavigationInjectedProps): StateProps => {
+  const { navigation } = ownProps
+  const transactionData = navigation.getParam('transactionData')
+  const isPaymentRequest = navigation.getParam('isPaymentRequest')
+  const { recipient } = transactionData
+  return {
+    recipient,
+    transactionData,
+    fullValidationRequired: navigation.getParam('fullValidationRequired'),
+    isValidRecipient: state.send.isValidRecipient,
+    isPaymentRequest,
+  }
+}
+
+type Props = StateProps & DispatchProps & WithTranslation & OwnProps
+
+export class ValidateRecipientAccount extends React.Component<Props, State> {
+  static navigationOptions = () => ({
+    ...headerWithBackButton,
+    headerTitle: i18n.t('sendFlow7:confirmAccountNumber.title'),
+  })
+
+  state: State = {
+    inputValue: '',
+    singleDigitInputValueArr: [],
+    isModalVisible: false,
+  }
+
+  componentDidUpdate = async () => {
+    const { isValidRecipient, isPaymentRequest, transactionData } = this.props
+    if (isValidRecipient) {
+      if (isPaymentRequest) {
+        navigate(Screens.PaymentRequestConfirmation, { transactionData })
+      } else {
+        navigate(Screens.SendConfirmation, { transactionData })
+      }
+    }
+  }
+
+  onPressContinue = () => {
+    const { inputValue, singleDigitInputValueArr } = this.state
+    const { recipient, fullValidationRequired } = this.props
+    const inputToValidate = fullValidationRequired ? inputValue : singleDigitInputValueArr.join('')
+    this.props.validateRecipientAddress(inputToValidate, fullValidationRequired, recipient)
+  }
+
+  onInputChange = (value: string) => {
+    this.setState({ inputValue: value })
+  }
+
+  onSingleDigitInputChange = (value: string, index: number) => {
+    const { singleDigitInputValueArr } = this.state
+    singleDigitInputValueArr[index] = value
+    this.setState({ singleDigitInputValueArr })
+  }
+
+  toggleModal = () => {
+    this.setState({ isModalVisible: !this.state.isModalVisible })
+  }
+
+  shouldShowClipboard = () => false
+
+  renderInstructionsAndInputField = () => {
+    const { t, recipient, fullValidationRequired } = this.props
+    const { inputValue, singleDigitInputValueArr } = this.state
+    const { displayName } = recipient
+
+    if (fullValidationRequired) {
+      return (
+        <View>
+          <Text style={styles.body}>
+            {t('confirmAccountNumber.1b', {
+              displayName,
+            })}
+          </Text>
+          <Text style={styles.body}>
+            {t('confirmAccountNumber.2b', {
+              displayName,
+            })}
+          </Text>
+          <Text style={styles.codeHeader}>{t('accountInputHeaderB')}</Text>
+          <CodeRow
+            status={CodeRowStatus.INPUTTING}
+            inputValue={inputValue}
+            inputPlaceholder={'0xf1b1d5a6e7728g309c4a025k122d71ad75a61976'}
```

Comment:
Why is a hardcoded placeholder needed here? If it is, can you make these placeholders constants above so it's clear it's intentional?

Reply:
Yup, good suggestion